### DISCUSSION TITLE
[docs] Kubernetes deployment

### DISCUSSION THREAD

**Discussion by robjuz** [osm-search/Nominatim#2412] [Show and tell]
I was working on a Helm chart for deploying Nominatim on Kubernetes

Maybe someone could review my work and if he likes it, add a note/link in the docs

https://github.com/robjuz/helm-charts/blob/master/charts/nominatim/README.md

**Comment by mtmail** [comment]
Thanks a lot. I can't read/understand much of Helm configuration but had a look at the values set, user names, which pages are used.

Building an image https://github.com/robjuz/postgresql-nominatim just to create a nominatim.so for later use is clever. The image says 3.7.2 but the main Chart.yaml (https://github.com/robjuz/helm-charts/blob/master/charts/nominatim/Chart.yaml) 3.7. There haven't been changes between .0 and .2 but it's something to watch out for. 100GB for flatnode file is plenty, 500GB for database storage will be too low for a full planet installation. 

I created a PR to add a link to the https://github.com/osm-search/Nominatim/blob/master/docs/admin/Installation.md page. Maybe the folks at https://github.com/mediagis/nominatim-docker would also add a link.

**Comment by robjuz** [comment]
@mtmail 
I deployed this with the europe extract. The flatnode file is huge (67 GB) and yes, the 500GB are probably not enough for the whole planet. This can be easily customized (thanks Helm)

```
df -h /nominatim/flatnode/
Filesystem                                                                                                                                                 Size  Used Avail Use% Mounted on
10.152.183.144:6789,10.152.183.74:6789,10.152.183.231:6789:/volumes/csi/csi-vol-3ab92919-e68b-11eb-89e9-929964496c43/4a9c7ff5-6325-4291-9439-efc063290bf1  100G   67G   34G  67% /nominatim/flatnode
```

```
df -h /bitnami/postgresql/
Filesystem      Size  Used Avail Use% Mounted on
/dev/rbd3       492G  373G  119G  76% /bitnami/postgresql
```
